packages/strapi-plugin-upload/admin/src/components/InputMedia/index.js

```diff
@@ -1,30 +1,74 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
-import CardPreview from '../CardPreview';
+import { getTrad } from '../../utils';
+import CardControl from '../CardControl';
+import CardControlWrapper from './CardControlWrapper';
+import CardPreviewWrapper from './CardPreviewWrapper';
+import EmptyInputMedia from './EmptyInputMedia';
+import EmptyText from './EmptyText';
+import IconUpload from './IconUpload';
+import InputFilePreview from './InputFilePreview';
 import InputModalStepper from '../../containers/InputModalStepper';
 import Name from './Name';
-import CardPreviewWrapper from './CardPreviewWrapper';
 import Wrapper from './Wrapper';
 
-const InputMedia = ({ label, onChange, name, attribute }) => {
+const InputMedia = ({ label, onChange, name, attribute, value, type }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [fileToDisplay, setFileToDisplay] = useState(0);
   const handleClickToggleModal = () => {
     setIsModalOpen(prev => !prev);
   };
+  const hasNoValue = Array.isArray(value) && value.length === 0;
+
+  const handleChange = v => {
+    onChange({ target: { name, type, value: v } });
+  };
+
+  const handleFilesNavigation = displayNext => {
+    if (displayNext && fileToDisplay === value.length - 1) {
+      setFileToDisplay(0);
+
+      return;
+    }
+
+    if (!displayNext && fileToDisplay === 0) {
+      setFileToDisplay(value.length - 1);
+    } else {
+      setFileToDisplay(prev => (displayNext ? prev + 1 : prev - 1));
+    }
+  };
+
+  const displaySlidePagination =
+    attribute.multiple && value.length > 1 ? ` (${fileToDisplay + 1}/${value.length})` : '';
 
   return (
     <Wrapper>
-      <Name htmlFor={name}>{label}</Name>
-      <CardPreviewWrapper onClick={handleClickToggleModal}>
-        <CardPreview url="https://images.pexels.com/photos/20787/pexels-photo.jpg?auto=compress&cs=tinysrgb&h=350" />
+      <Name htmlFor={name}>{`${label}${displaySlidePagination}`}</Name>
+
+      <CardPreviewWrapper>
+        <CardControlWrapper>
+          <CardControl color="#9EA7B8" type="plus" onClick={() => handleClickToggleModal(value)} />
```

Comment:
I think the handleClickToggleModal() should also be on the EmptyInputMedia to open the modal on the entire empty input

Reply:
Indeed